### PR TITLE
Fix byte[] equality issue for MockByteArraySparkeyReader

### PR DESCRIPTION
**Pull request by clairemcginty**
MockByteArraySparkeyReader was not working properly since two `Array[Byte]`s containing the same bytes produce a different hashCode every time they're deserialized -- thus, the `Map#get` operation was throwing a `NoSuchElementException` even if it contained an Array[Byte] key matching the requested key bytes.

This solution wraps map keys in a ByteBuffer, which is safe to use as a Map key. It also returns `null` if the key doesn't exist in the map rather than throwing a `NoSuchElementException`, which matches the prior Sparkey behavior

### CODE REVIEW

File: scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/SparkeyIO.scala [hunk 1]
```diff
@@ -33,7 +33,7 @@ import scala.collection.mutable
 import scala.jdk.CollectionConverters._
 
 /** Special version of [[com.spotify.scio.io.ScioIO]] for use with sparkey methods. */
-case class SparkeyTestIO[T] private (path: String) extends TestIO[T] {
+private[sparkey] case class SparkeyTestIO[T](path: String) extends TestIO[T] {
```
Comment (clairemcginty) [inline scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/SparkeyIO.scala:36]: It seemed confusing to support both this constructor as well as SparkeyIO#apply, [below](https://github.com/spotify/scio/blob/v0.14.0/scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/SparkeyIO.scala#L48).

File: scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/instances/MockSparkeyReader.scala [hunk 2]
```diff
@@ -74,7 +75,10 @@ case class MockByteArrayEntry(k: Array[Byte], v: Array[Byte])
 
 case class MockByteArraySparkeyReader(data: Map[Array[Byte], Array[Byte]])
     extends MockSparkeyReader {
-  override def getAsByteArray(key: Array[Byte]): Array[Byte] = data(key)
+  private lazy val internal = data.map { case (k, v) => (ByteBuffer.wrap(k), v) }
```
Comment (kellen) [inline scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/instances/MockSparkeyReader.scala:78]: Using ByteString might be better here.

**Review by kellen — APPROVED** [review]
D'oh. Thanks for fixing.

File: scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/instances/MockSparkeyReader.scala (hunk 2 above)
Comment (clairemcginty) [reply scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/instances/MockSparkeyReader.scala:78]: true, although that bring in a protobuf dependency to scio-sparkey where there was none before

probably not a big deal

Comment (RustedBones) [reply scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/instances/MockSparkeyReader.scala:78]: why not using scala `Seq[Byte]` or `Vector[Byte]` ?

Comment (clairemcginty) [reply scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/instances/MockSparkeyReader.scala]: good idea! updated

**Review by RustedBones — APPROVED** [review]

**Pull request merged**